vm/systemSmartContracts/eei.go

```diff
@@ -272,6 +280,63 @@ func (host *vmContext) Transfer(
 	return nil
 }
 
+// ProcessBuiltInFunction will execute process if sender and destination is same shard/sovereign
+func (host *vmContext) ProcessBuiltInFunction(
+	destination []byte,
+	sender []byte,
+	value *big.Int,
+	input []byte,
+	gasLimit uint64,
+) error {
+	isIntraBuiltIn, vmInput := host.isIntraShardBuiltInCall(destination, sender, value, input, gasLimit)
+	if !isIntraBuiltIn {
+		return host.Transfer(destination, sender, value, input, gasLimit)
+	}
+
+	vmOutput, err := host.blockChainHook.ProcessBuiltInFunction(vmInput)
+	if err != nil {
+		return err
+	}
+	if vmOutput.ReturnCode != vmcommon.Ok {
+		return errors.New(vmOutput.ReturnMessage)
+	}
+
+	// add the SCR for the builtin function
+	return host.Transfer(destination, sender, value, input, gasLimit)
```

Comment:
is this necessary? do we have a SCR generated intrashard?

Reply:
yes. we generate SCRs for all SC to SC calls - as we want to index those to explorer.

Reply:
ok